### PR TITLE
Add support for instant vectors to vector_selector

### PR DESCRIPTION
**Pull request by JamesGuthrie**
An instant vector query is defined by start == end and bucket_width == 0

### CODE REVIEW

**Comment by JamesGuthrie** [comment]
Note: This PR builds on top of the commit in #76, once that's merged I'll rebase and merge this change.

File: src/aggregates/vector_selector.rs
Comment (bitwalker) [inline src/aggregates/vector_selector.rs:173]: **Issue:** Did you intend to use bitwise XOR here? It will do the wrong thing if both expressions are true (i.e. it will return false, even if one of the expressions is true). I think you probably meant to use logical OR `||` here, but perhaps I'm misunderstanding.

Comment (JamesGuthrie) [reply src/aggregates/vector_selector.rs:173]: From what I can tell, this is logical XOR: https://doc.rust-lang.org/reference/types/boolean.html#logical-xor

Comment (bitwalker) [reply src/aggregates/vector_selector.rs:173]: On booleans, bitwise XOR and logical XOR are equivalent, but in general `^` is the bitwise XOR operator, as shown [here](https://doc.rust-lang.org/book/appendix-02-operators.html). I was more confused by the use of XOR here, than the fact that it's a bitwise operator.

Comment (JamesGuthrie) [reply src/aggregates/vector_selector.rs:173]: Yes, it is intentional. When invoked as an instant vector, we expect that `bucket_width == 0`, _and_ `start_time == end_time`.

When not invoked as an instant vector, we don't have any constraints on these parameters.

I could have written it as:

```rust
if (bucket_width == 0 && start_time != end_time) || (start_time == end_time && bucket_width != 0) {
  ...
} 
```

Which is equivalent to the expression used here.

Comment (bitwalker) [reply src/aggregates/vector_selector.rs:173]: Makes sense. I think why reading this was confusing at first glance is that it implicitly relies on classifying the vector type as instant if either `bucket_width == 0` or `start_time == end_time`, even though a vector is only instant if both conditions are true. Packing that all into an XOR is elegant, but had me re-reading it multiple times to make sure I was understanding the logic correctly. Might be worth adding a comment, but I'll leave that up to you if you think it is sufficiently clear - we can chalk it up to me not reading the docs carefully enough

Comment (JamesGuthrie) [reply src/aggregates/vector_selector.rs:173]: Hmmm... I agree that it's not super explicit. I will take a look at how to make it more explicit tomorrow.

File: src/aggregates/vector_selector.rs
Comment (cevian) [inline src/aggregates/vector_selector.rs:27]: Hmm for range vectors there is no lookback. You always consider the entire bucket. In other words the lookback=bucket_width

File: src/aggregates/vector_selector.rs
Comment (cevian) [reply src/aggregates/vector_selector.rs:173]: yeah lets make the logic here more clear

**Review by cevian — APPROVED** [review]
LGTM just asked for some doc fixes

**Review by bitwalker — APPROVED** [review]

File: src/aggregates/vector_selector.rs
Comment (JamesGuthrie) [reply src/aggregates/vector_selector.rs:27]: Slowly my mental model of this is starting to take shape. Thanks!

File: src/aggregates/vector_selector.rs
Comment (JamesGuthrie) [reply src/aggregates/vector_selector.rs]: I looked into this again, I think that it was just incorrect to impose this condition. I've removed it.

**Review by bitwalker — APPROVED** [review]

**Pull request merged**
